src/runtime_src/core/tools/common/ReportAieShim.cpp

```diff
@@ -17,224 +17,26 @@
-          channel.put("id", id++);
-          channel.put("channel_status", node.second.data());
-          channel.put("queue_size", queue_size->second.data());
-          channel.put("queue_status", queue_status->second.data());
-          channel.put("current_bd", current_bd->second.data());
-          queue_size++;
-          queue_status++;
-          current_bd++;
-          mm2s_array.push_back(std::make_pair("", channel));
-        }
-        ishim.add_child("dma.mm2s.channel", mm2s_array);
-
-        boost::property_tree::ptree s2mm_array;
-        queue_size = oshim.get_child("dma.queue_size.s2mm").begin();
-        queue_status = oshim.get_child("dma.queue_status.s2mm").begin();
-        current_bd = oshim.get_child("dma.current_bd.s2mm").begin();
-        id = 0;
-        for (auto& node : oshim.get_child("dma.channel_status.s2mm")) {
-          boost::property_tree::ptree channel;
-          channel.put("id", id++);
-          channel.put("channel_status", node.second.data());
-          channel.put("queue_size", queue_size->second.data());
-          channel.put("queue_status", queue_status->second.data());
-          channel.put("current_bd", current_bd->second.data());
-          queue_size++;
-          queue_status++;
-          current_bd++;
-          s2mm_array.push_back(std::make_pair("", channel));
-        }
-        ishim.add_child("dma.s2mm.channel", s2mm_array);
-      }
-
-      if(oshim.find("lock") != oshim.not_found())
-        addnodelist("lock", "locks", oshim, ishim);
-
-      boost::property_tree::ptree module_array;
-      for (auto& node : oshim.get_child("errors", empty_pt)) {
-        boost::property_tree::ptree module;
-        module.put("module",node.first);
-        boost::property_tree::ptree type_array;
-        for (auto& type : node.second) {
-          boost::property_tree::ptree enode;
-          enode.put("name",type.first);
-          std::string val;
-          for (auto& value: type.second)
-            val +=(val.empty()?"":", ")+ value.second.data();
-          enode.put("value", val);
-          type_array.push_back(std::make_pair("", enode));
-        }
-        module.add_child("error", type_array);
-        module_array.push_back(std::make_pair("", module));
-      }
-
-      if(oshim.find("errors") != oshim.not_found())
-        ishim.add_child("errors", module_array);
-
-      addnodelist("event", "events", oshim, ishim);
-      tile_array.push_back(std::make_pair("tile"+std::to_string(col),ishim));
-    }
-    pt.add_child("tiles",tile_array);
-  } catch (const std::exception& ex){
-    pt.put("error_msg", (boost::format("%s %s") % ex.what() % "found in the AIE shim"));
-  }
-
+  xrt::device device(_pDevice->get_device_id());
+  boost::property_tree::ptree pt_shim;
+  std::stringstream ss;
+  ss << device.get_info<xrt::info::device::aie_shim>();
+  boost::property_tree::read_json(ss, pt_shim);
+
+  // There can only be 1 root node
+  pt.add_child("electrical", pt_shim);

```

Comment:
This shouldn't be electrical.  I think, this should be tiles.

Reply:
OOps !! Fixed. Thanks